src/mpid/ch4/netmod/ofi/ofi_types.h

```diff
@@ -555,4 +564,28 @@ extern MPIDI_OFI_huge_recv_list_t *MPIDI_posted_huge_recv_tail;
 extern int MPIR_Datatype_init_names(void);
 extern MPIDI_OFI_capabilities_t MPIDI_OFI_caps_list[MPIDI_OFI_NUM_SETS];
 
+/* Build uniform mr_key using contextid, key type and uniq ID passed */
+MPL_STATIC_INLINE_PREFIX uint64_t MPIDI_OFI_rma_key_pack(MPIR_Context_id_t ctx, int key_type,
+                                                         uint64_t key_id)
+{
+    uint64_t rma_key = 0;
+    rma_key = ((((uint64_t) (MPIR_CONTEXT_READ_FIELD(PREFIX, ctx))) << (MPIDI_Global.context_shift))
```

Comment:
This is almost impossible to understand. Can it be simplified? Maybe into multiple operations?

Reply:
@raffenet yes, it looks like a lisp code. I've separete it into several parts and add comments.